### PR TITLE
[doc] Document Top Earlgrey Parameterisations

### PR DESCRIPTION
**Pull request by lenary**
At the moment, we only have one top level (top_earlgrey) and therefore only
one parameterisation of various peripherals (alert handler, pinmux,
PLIC).

The docs, until today, have been generated from the "default"
parameterisations, which have not matched the top earlgrey
parameterisations, which has made writing software against the hardware
difficult and error-prone.

This commit changes the docs to use the earlgrey parameterisations, and
documents the parameterisations in the parameter tables. Should we gain
other top levels, we should revisit how this is generated, but we knew
that already, and this makes the software developers lives much easier.

### CODE REVIEW

File: hw/ip/alert_handler/doc/_index.md [hunk 1]
```diff
@@ -902,32 +902,28 @@ not be triggered by this testing mechanism.
 
 ## Register Table
 
-The below register description can be generated with the `reg_alert_handler.py`
-script. The reason for having yet another script for register generation is
-that the alert handler is configurable for the number of alert sources
-(similar to the rv_plic design).
+The register description below matches the instance in Top Earlgrey.
```
Comment (imphil) [inline hw/ip/alert_handler/doc/_index.md]: "Top Earlgrey" -> "earlgrey top level design" (or Earl Grey, we don't really have a consistent naming convention)

File: hw/ip/alert_handler/doc/_index.md [hunk 2]
```diff
@@ -902,32 +902,28 @@ not be triggered by this testing mechanism.
...
-The register description below may not match with the actual instance in Top
-Earlgrey, since the alert handler is generated by the topgen tool so that the
-number of alert sources matches.
+A similar register description can be generated with the `reg_alert_handler.py`
+script. The reason for having yet another script for register generation is that
+the alert handler is configurable for the number of alert sources (similar to
+the rv_plic design).
 
-In order to generate the register file below, from `hw/ip/alert_handler/doc`:
+In order to generate the register file for 4 alert sources, from
```
Comment (imphil) [inline hw/ip/alert_handler/doc/_index.md]: 4 -> four

File: hw/ip/pinmux/doc/_index.md [hunk 3]
```diff
@@ -125,7 +125,8 @@ The configuration can then not be altered anymore unless the system is reset.
 
 ## Register Table
 
-The layout of the register table can change, based on the parameterization.
-The register table below is an example and has been generated using the default parameters.
+The register description below matches the instance in Top Earlgrey.
```
Comment (imphil) [inline hw/ip/pinmux/doc/_index.md]: as above

File: util/build_docs.py [hunk 4]
```diff
@@ -49,16 +49,19 @@
     # Pre-generate register and hwcfg fragments from these files.
     "hardware_definitions": [
         "hw/ip/aes/data/aes.hjson",
-        "hw/ip/alert_handler/data/alert_handler.hjson",
+        # "hw/ip/alert_handler/data/alert_handler.hjson",
```
Comment (imphil) [inline util/build_docs.py]: Please add a comment somewhere what the commented-out pieces mean, or remove them.

File: hw/ip/alert_handler/doc/_index.md (hunk 1 above)
Comment (lenary) [reply hw/ip/alert_handler/doc/_index.md]: Done

File: hw/ip/alert_handler/doc/_index.md (hunk 2 above)
Comment (lenary) [reply hw/ip/alert_handler/doc/_index.md]: Done

File: hw/ip/pinmux/doc/_index.md (hunk 3 above)
Comment (lenary) [reply hw/ip/pinmux/doc/_index.md]: Done

File: util/build_docs.py (hunk 4 above)
Comment (lenary) [reply util/build_docs.py]: Done

**Comment by eunchan** [comment]
LGTM. Could you please also link the issue here and also the commit message?'
Putting the TODO into the doc will be good also.

**Comment by lenary** [comment]
Closes #1068, #1496

**Comment by silvestrst** [comment]
@lenary thanks, this is great stuff!

File: util/build_docs.py [hunk 5]
```diff
@@ -47,18 +47,24 @@
...
         "hw/ip/entropy_src/data/entropy_src.hjson",
         "hw/ip/flash_ctrl/data/flash_ctrl.hjson",
         "hw/ip/gpio/data/gpio.hjson",
         "hw/ip/hmac/data/hmac.hjson",
         "hw/ip/i2c/data/i2c.hjson",
         "hw/ip/nmi_gen/data/nmi_gen.hjson",
         "hw/ip/padctrl/data/padctrl.hjson",
-        "hw/ip/pinmux/data/pinmux.hjson",
-        "hw/ip/rv_plic/data/rv_plic.hjson",
+        # "hw/ip/pinmux/data/pinmux.hjson",
```
Comment (sjgitty) [inline util/build_docs.py]: if we're OK with these changes, we should remove the comments.

Comment (msfschaffner) [reply util/build_docs.py]: agreed

File: hw/ip/alert_handler/doc/_index.md [hunk 6]
```diff
@@ -128,13 +128,13 @@ configuration, and hence these parameters are placed into a package as
...
-Localparam     | Default (Max)         | Description
----------------|-----------------------|---------------
-`NAlerts`      | 8 (248)               | Number of alert instances. Maximum number bounded by LFSR implementation that generates ping timing.
-`EscCntWidth`  | 32 (32)               | Width of the escalation counters in bit.
-`AccuCntWidth` | 16 (32)               | Width of the alert accumulation counters in bit.
-`AsyncOn`      | '0 (2^`NAlerts`-1)    | This is a bit array specifying whether a certain alert sender / receiver pair goes across an asynchronous boundary or not.
-`LfsrSeed`     | '1 (2^31-1)           | Seed for the LFSR timer, must be nonzero.
+Localparam     | Default (Max)         | Top Earlgrey | Description
+---------------|-----------------------|--------------|---------------
+`NAlerts`      | 8 (248)               | 1            | Number of alert instances. Maximum number bounded by LFSR implementation that generates ping timing.
```
Comment (msfschaffner) [inline hw/ip/alert_handler/doc/_index.md]: For the time being this is probably ok, but eventually we should discuss how we could infer these from the configuration.

Comment (lenary) [reply hw/ip/alert_handler/doc/_index.md]: Oh sure, I think this should eventually be generated too, but I felt that was too much for this PR.

**Pull request merged**